lms/djangoapps/certificates/signals.py

```diff
@@ -81,17 +82,38 @@ def _listen_for_id_verification_status_changed(sender, user, **kwargs):  # pylin
 
 def fire_ungenerated_certificate_task(user, course_key, expected_verification_status=None):
     """
-    Helper function to fire un-generated certificate tasks
-
-    The 'mode_is_verified' query is copied from the GeneratedCertificate model,
-    but is done here in an attempt to reduce traffic to the workers.
-    If the learner is verified and their cert has the 'unverified' status,
-    we regenerate the cert.
+    Helper function to fire certificate generation task.
+    Auto-generation of certificates is available for following course modes:
+        1- VERIFIED
+        2- CREDIT_MODE
+        3- PROFESSIONAL
+        4- NO_ID_PROFESSIONAL_MODE
+
+    NOTE: Purpose of restricting other course modes (HONOR and AUDIT) from auto-generation is to reduce
+    traffic to workers.
     """
+
+    # course enrollment modes allowed for auto-generation of certificates
+    allowed_enrollement_modes_list = [
```

Comment:
Intention of putting up this list is to decouple **enrollment modes with verified certificate** from **enrollment modes allowed for auto-generation of certificates**.